certora/specs/LiquidateHealthStatus.spec

```diff
@@ -0,0 +1,399 @@
+    - account checked != liquidator and account checked != violator
+    - account checked == liquidator and account checked != violator and:
+        - debt socialization disabled
+        - debt socialization enabled 
+These cases are handled separately:
+    - account checked == violator:
+        if this account was healthy before the call does nothing. This is not
+        only easy to see manually but we prove this in 
+        checkLiquidation.healthy() in Liquidation.spec
+    - liquidator  == violator:
+        In this case the call reverts which is easy to check but also
+        proved in liquidate_mustRevert in Liquidation.spec
+*/
+
+// passing: https://prover.certora.com/output/65266/132c942ca2a2463b84e15b77becdfa11/?anonymousKey=431faa27dd7b649306de7e37067b7a75e57271f8
+rule liquidateAccountsStayHealthy_liquidator_no_debt_socialization {
+    env e;
+    address account;
+    address[] collaterals = evc.getCollaterals(e, account);
+    require collaterals.length <= 2; // loop bound
+    require oracleAddress != 0; 
+    // Vault cannot be a user of itself
+    require account != currentContract;
+    // Vault should not be used as a collateral
+    require collaterals[0] != currentContract;
+    require collaterals[1] != currentContract;
+    // not sure the following 4 are really needed
+    require account != erc20;
+    require account != oracleAddress;
+    require account != evc;
+    require account != unitOfAccount;
+
+    require LTVConfigAssumptions(e, getLTVConfig(e, ETokenA));
+    require LTVConfigAssumptions(e, getLTVConfig(e, ETokenB));
+    // Collaterals must be ETokens
+    if (collaterals.length > 0) {
+        require collaterals[0] == ETokenA;
+    }
+    if (collaterals.length > 1) {
+        require collaterals[1] == ETokenB;
+    }
+
+    address violator;
+    address collateral;
+    uint256 repayAssets;
+    uint256 minYieldBalance;
+
+    // disable debt socialization
+    require !hasDebtSocialization();
+
+    // initialize checked accounts to 0
+    require accountToCheckGhost == 0; // account checked in initialize
+    require collateralTransferCheckedAccount == 0;
+
+    // account eq liquidator case
+    require collateral == ETokenA || collateral == ETokenB;
+    address liquidator = actualCaller(e);
+    require account == liquidator;
+    require violator != liquidator;
+
+    bool healthyBefore = checkLiquidityReturning(e, account, collaterals);
+    currentContract.liquidate(e, violator, collateral, repayAssets, minYieldBalance);
+
+    // replace the real call path involving the EVC calling back into the
+    // vault with a direct call on checkAccountStatus from the vault
+
+    if(accountToCheckGhost != 0) {
+        currentContract.checkAccountStatus(e, accountToCheckGhost, collaterals);
+    }
+    if(collateralTransferCheckedAccount != 0) {
+        currentContract.checkAccountStatus(e, collateralTransferCheckedAccount, collaterals);
```

Comment:
I think you are right and this can be deleted (for the repeated pattern below as well). I made this change locally and am just waiting on the runs to make sure this still passes.